src/main/kotlin/io/lenses/jdbc4/row/rows.kt

```diff
@@ -1,13 +1,22 @@
 package io.lenses.jdbc4.row
 
 import org.apache.avro.generic.GenericData
+import java.sql.SQLException
 
 /**
  * An implementation of [Row] that uses a static list of values
  * provided at construction time.
  */
 class ListRow(private val array: List<Any?>) : ConvertingRow() {
-  override fun getObject(index: Int): Any? = array[index - 1]
+  override fun getObject(index: Int): Any? = try {
+    if (index > array.size) {
```

Comment:
This is a problem even on `SHOW FUNCTIONS` because the column set is 3 but some records because the function does not have a specific return type it will fail with out of bounds so if you want to read via .toList it will fail

Reply:
Actually that was not it. but during test i can see the issue where a row has less columns because they are null

Reply:
@sksamuel We need a solution for this.
`SHOW FUNCTIONS` metadata is name/description/return_type

However there is no value for return_type for quite a few functions. this means this code fails
on ` rs.getString("return_type")` with index out of bounds since there is no date returned.
```
 val rs = stmt.executeQuery("SHOW FUNCTIONS")
      val result = mutableListOf<List<Any>>()
      while (rs.next()) {
        val name = rs.getString("name")
        val desc = rs.getString("description")
        val type = rs.getString("return_type")
        val row = listOf(name, desc, type)
        result.add(row)
      }
```